[1.10] panic in logging

```
panic: runtime error: index out of range

goroutine 5258 [running]:
bytes.(*Buffer).WriteByte(0xc8208c9dc0, 0x15d6822, 0x0, 0x0)
	/usr/local/go/src/bytes/buffer.go:233 +0x80
github.com/docker/docker/pkg/jsonlog.ffjsonWriteJSONString(0xc8208c9dc0, 0x15d68b0, 0x6)
	/go/src/github.com/docker/docker/pkg/jsonlog/jsonlog_marshalling.go:179 +0x100
github.com/docker/docker/pkg/jsonlog.(*JSONLogs).MarshalJSONBuf(0xc820fc6ca0, 0xc8208c9dc0, 0x0, 0x0)
	/go/src/github.com/docker/docker/pkg/jsonlog/jsonlogbytes.go:39 +0x138
github.com/docker/docker/daemon/logger/jsonfilelog.(*JSONFileLogger).Log(0xc8214d63c0, 0xc8207fd810, 0x0, 0x0)
	/go/src/github.com/docker/docker/daemon/logger/jsonfilelog/jsonfilelog.go:98 +0x1b5
github.com/docker/docker/daemon/logger.(*Copier).copySrc(0xc821dd6300, 0x15d68b0, 0x6, 0x7f495aa54c38, 0xc821545270)
	/go/src/github.com/docker/docker/daemon/logger/copier.go:53 +0x4b6
created by github.com/docker/docker/daemon/logger.(*Copier).Run
	/go/src/github.com/docker/docker/daemon/logger/copier.go:38 +0x127
```

```
docker version; docker info
Client:
 Version:      1.9.1
 API version:  1.21
 Go version:   go1.4.3
 Git commit:   a34a1d5
 Built:        Fri Nov 20 17:56:04 UTC 2015
 OS/Arch:      linux/amd64

Server:
 Version:      1.10.0-dev
 API version:  1.22
 Go version:   go1.5.2
 Git commit:   3df4a51
 Built:        2015-12-31T06:55:37.322373109+00:00
 OS/Arch:      linux/amd64
Containers: 3
Images: 12
Server Version: 1.10.0-dev
Storage Driver: overlay
 Backing Filesystem: extfs
Execution Driver: native-0.2
Logging Driver: json-file
Kernel Version: 4.2.5
Operating System: Ubuntu 15.10 (containerized)
CPUs: 4
Total Memory: 15.4 GiB
Name: inotmac
ID: HMGD:MCSR:XSNO:I7OW:2H7X:KE3J:XIUK:FEOU:SR4J:PVBZ:B3KW:2C6V
Debug mode (server): true
 File Descriptors: 20
 Goroutines: 38
 System Time: 2015-12-31T08:35:25.482973866-07:00
 EventsListeners: 1
 Init SHA1: 
 Init Path: /home/darren/src/docker/bundles/latest/binary/docker-1.10.0-dev
 Docker Root Dir: /var/lib/docker
Username: ibuildthecloud
Registry: https://index.docker.io/v1/
WARNING: No swap limit support
```

--- a/daemon/logger/jsonfilelog/jsonfilelog.go
+++ b/daemon/logger/jsonfilelog/jsonfilelog.go
@@ -90,6 +90,8 @@ func (l *JSONFileLogger) Log(msg *logger.Message) error {
 	if err != nil {
 		return err
 	}
+	l.mu.Lock()
+	defer l.mu.Unlock()
 	err = (&jsonlog.JSONLogs{
 		Log:      append(msg.Line, '\n'),
 		Stream:   msg.Source,

